core/src/main/java/com/backblaze/b2/client/B2StorageClientWebifierImpl.java

```diff
@@ -423,11 +426,15 @@ public B2FileVersion getFileInfo(B2AccountAuthorization accountAuth,
 
     @Override
     public B2FileVersion getFileInfoByName(B2AccountAuthorization accountAuth, B2GetFileInfoByNameRequest request) throws B2Exception {
-        return webApiClient.postJsonReturnJson(
-                makeGetFileInfoByNameUrl(accountAuth, request.getBucketName(), request.getFileName()),
-                makeHeaders(accountAuth),
-                request,
-                B2FileVersion.class);
+        B2HeadersImpl.Builder builder = B2HeadersImpl.builder();
+        webApiClient.head(makeGetFileInfoByNameUrl(accountAuth, request.getBucketName(), request.getFileName()),
+                makeHeaders(accountAuth)).forEach((key, value) -> builder.set(key, value));
+
+        B2Headers headers = builder.build();
+
+        return new B2FileVersion(headers.getValueOrNull(FILE_ID), headers.getValueOrNull(FILE_NAME),
```

Comment:
i found this easier to read when i rewrote it with a couple of temporary variables (and pretended that head() returned a B2Headers object).

changing it so that the arguments to B2FileVersion were one per line made it obvious that we're not yet able to fill in the 'action' of the B2FileVersion.   i think that you can hard-code "upload" there for now, as i've done below because we can't getFileInfoByName for hidden files or large files that are still being uploaded.  (I will confirm that with some co-workers and we may eventually teach the B2 service to return the action so it doesn't have to be hard-coded here.)

    @Override
    public B2FileVersion getFileInfoByName(B2AccountAuthorization accountAuth, B2GetFileInfoByNameRequest request) throws B2Exception {
        final String url = makeGetFileInfoByNameUrl(accountAuth, request.getBucketName(), request.getFileName());
        B2Headers headers = webApiClient.head(url, makeHeaders(accountAuth));

        return new B2FileVersion(headers.getValueOrNull(FILE_ID),
                headers.getValueOrNull(FILE_NAME),
                headers.getContentLength(),
                headers.getContentType(),
                headers.getContentSha1OrNull(),
                headers.getB2FileInfo(),
                "upload",
                Long.parseLong(headers.getValueOrNull(UPLOAD_TIMESTAMP)));
    }